File: Year3/Semester 1/Programare pentru dispozitive mobile/Projects/MobileApplicationProject/app/src/main/java/com/example/mobileapplicationproject/data/interfaces/GoalsPage.kt
package com.example.mobileapplicationproject.data.interfaces

import com.example.mobileapplicationproject.data.components.GoalRepo
import com.example.mobileapplicationproject.data.model.GoalViewModel
import androidx.lifecycle.viewmodel.compose.viewModel
import androidx.navigation.compose.*
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import com.example.mobileapplicationproject.data.components.DeleteConfirmationDialog
import com.example.mobileapplicationproject.data.components.GoalCreationScreen
import com.example.mobileapplicationproject.data.components.GoalUpdateScreen

@Composable
fun GoalsPage()
{
    val viewModel: GoalViewModel = viewModel()
    val navController = rememberNavController()

    NavHost(
        navController = navController,
        startDestination = "main"
    ) {
        composable("main") {
            GoalRepo(viewModel, navController)
        }
        composable("addGoal") {
            GoalCreationScreen(
                viewModel = viewModel,
                onGoalCreated = { newGoal ->
                    // Handle the newly created goal here (e.g., save it in the ViewModel)
                    viewModel.addGoal(newGoal)
                    // You can navigate back to the main screen or any other screen as needed
                    navController.navigate("main")
                }
            )
        }
//        composable("updateGoal/{goalTitle}") { backStackEntry ->
//            val arguments = requireNotNull(backStackEntry.arguments)
//            val goalTitle = arguments.getString("goalTitle")
//            val goal = goalTitle?.let { viewModel.getGoalByTitle(it) }
//
//            if (goal != null) {
//                GoalUpdateScreen(
//                    viewModel = viewModel,
//                    goal = goal,
//                    onUpdate = { updatedGoal ->
//                        // Handle the updated goal here (e.g., update it in the ViewModel)
//                        viewModel.updateGoal(updatedGoal)
//                        // You can navigate back to the main screen or any other screen as needed
//                        navController.navigate("main")
//                    }
//                )
//            } else {
//                Text("Goal not found")
//            }
//        }
//        composable("deleteGoal/{goalTitle}") { backStackEntry ->
//            val arguments = requireNotNull(backStackEntry.arguments)
//            val goalTitle = arguments.getString("goalTitle")
//            val goal = goalTitle?.let { viewModel.getGoalByTitle(it) }
//
//            if (goal != null) {
//                DeleteConfirmationDialog(
//                    goal = goal,
//                    onConfirmDelete = {
//                        // Handle goal deletion here (e.g., delete it in the ViewModel)
//                        viewModel.deleteGoal(goal.title)
//                        // You can navigate back to the main screen or any other screen as needed
//                        navController.navigate("main")
//                    },
//                    onDismiss = {
//                        // Handle dismissal of the dialog (e.g., navigate back to the main screen)
//                        navController.navigate("main")
//                    }
//                )
//            } else {
//                Text("Goal not found")
//            }
//        }
    }
}

File: Year3/Semester 1/Programare pentru dispozitive mobile/Projects/new_flutter_attempt/lib/interfaces/goals_page.dart
import 'package:flutter/material.dart';
import '../components/goal_display_list.dart';
import '../model/goals_view_model.dart';
import 'goal_creation_screen.dart';
import 'goal_update_screen.dart';
import 'goal_delete_screen.dart';

class MainScreen extends StatefulWidget {
  @override
  _MainScreenState createState() => _MainScreenState();
}

class _MainScreenState extends State<MainScreen> {
  late GoalViewModel viewModel;

  @override
  void initState() {
    super.initState();
    // Initialize your view model
    viewModel = GoalViewModel(); // Replace this with your actual initialization logic
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('Goals App'),
      ),
      body: GoalList(
        viewModel: viewModel,
        onEditClick: (goal) {
          // Handle edit click: Navigate to goal update screen
          Navigator.push(
            context,
            MaterialPageRoute(
              builder: (context) => GoalUpdateScreen(
                viewModel: viewModel,
                goal: goal,
                onUpdate: (updatedGoal) {
                  // Handle the updated goal
                  viewModel.updateGoal(updatedGoal);
                  // Close the update screen and refresh the goal list
                  Navigator.pop(context);
                  setState(() {});
                },
              ),
            ),
          );
        },
        onDeleteClick: (goal) {
          // Handle delete click: Show delete confirmation dialog
          showDialog(
            context: context,
            builder: (context) => DeleteConfirmationDialog(
              onConfirmDelete: () {
                // Handle goal deletion
                viewModel.deleteGoal(goal.title);
                // Close the delete confirmation dialog and refresh the goal list
                Navigator.pop(context);
                setState(() {});
              },
              onDismiss: () {
                // Close the delete confirmation dialog
                Navigator.pop(context);
              },
            ),
          );
        },
      ),
      floatingActionButton: FloatingActionButton(
        onPressed: () {
          // Navigate to the "Create Goal" screen
          Navigator.push(
            context,
            MaterialPageRoute(
              builder: (context) => GoalCreationScreen(
                viewModel: viewModel,
                onGoalCreated: (newGoal) {
                  // Handle the newly created goal
                  viewModel.addGoal(newGoal);
                  // Close the create goal screen and refresh the goal list
                  Navigator.pop(context);
                  setState(() {});
                },
              ),
            ),
          );
        },
        child: Icon(Icons.add),
      ),
    );
  }
}
